qiskit/transpiler/passes/optimization/hoare_opt.py

```diff
@@ -212,11 +215,12 @@ def _traverse_dag(self, dag):
             remove_ctrl, new_dag, qb_idx = self._remove_control(gate, ctrlvar, trgtvar)
 
             if remove_ctrl:
-                dag.substitute_node_with_dag(node, new_dag)
-                gate = gate.base_gate
-                node.op = gate.to_mutable()
-                node.name = gate.name
-                node.qargs = tuple((ctrlqb + trgtqb)[qi] for qi in qb_idx)
+                # We are replacing a node by a new node over a smaller number of qubits.
+                # This can be done using substitute_node_with_dag, which furthermore returns
+                # a mapping from old node ids to new nodes.
+                mapped_nodes = dag.substitute_node_with_dag(node, new_dag)
+                node = list(mapped_nodes.values())[0]
```

Comment:
I think this should be slightly faster as you don't need to generate the entire list 🙂 
```suggestion
                node = next(iter(mapped_nodes.values()))
```

Reply:
Done in [d819a42](https://github.com/Qiskit/qiskit/pull/13083/commits/d819a424df9297eb2792dce77ef4079f9a9f8c17). Indeed, this form is better.